tools/ob_admin/backup_tool/ob_admin_backup_validation_task.cpp

```diff
@@ -0,0 +1,4804 @@
+        ret = OB_ALLOCATE_MEMORY_FAILED;
+        STORAGE_LOG(WARN, "failed to alloc backup::ObBackupTabletMeta", K(ret));
+      } else if (FALSE_IT(tablet_meta_.at(i) = new (alc_ptr) backup::ObBackupTabletMeta())) {
+      } else if (OB_FAIL(ObAdminBackupValidationUtil::read_tablet_meta(
+                     backup_set_attr->backup_set_store_->get_backup_set_dest(),
+                     *tablet_attr->tablet_meta_index_, tablet_attr->data_type_, *tablet_meta_.at(i),
+                     ctx_))) {
+        STORAGE_LOG(WARN, "failed to read tablet meta", K(ret), K(backup_set_id),
+                    K(tablet_attr->tablet_meta_index_), K(tablet_attr->data_type_));
+      } else if (!tablet_meta_.at(i)->tablet_meta_.ha_status_.is_none()) {
+        ret = OB_ERR_UNEXPECTED;
+        STORAGE_LOG(WARN, "tablet meta is not complete", K(ret), K(backup_set_id),
+                    K(*tablet_meta_.at(i)));
+      }
+    }
+    if (OB_SUCC(ret)) {
+      if (2 == current_group.count()) {
+        // major/minor same tablet
+        if (tablet_meta_.at(0)->tablet_id_ != tablet_meta_.at(1)->tablet_id_) {
+          ret = OB_ERR_UNEXPECTED;
+          STORAGE_LOG(WARN, "tablet id is not same", K(ret), K(tablet_meta_.at(0)->tablet_id_),
+                      K(tablet_meta_.at(1)->tablet_id_));
+        } else if (tablet_meta_.at(0)->tablet_meta_.clog_checkpoint_scn_
+                   > tablet_meta_.at(1)->tablet_meta_.clog_checkpoint_scn_) {
+          ret = OB_ERR_UNEXPECTED;
+          STORAGE_LOG(WARN, "minor tablet clog checkpoint scn is larger than major tablet", K(ret),
+                      K(tablet_meta_.at(0)->tablet_meta_.clog_checkpoint_scn_),
+                      K(tablet_meta_.at(1)->tablet_meta_.clog_checkpoint_scn_));
+        } else {
+          const int64_t major_snapshot_version
+              = tablet_meta_.at(1)->tablet_meta_.report_status_.merge_snapshot_version_;
+          const int64_t minor_snapshot_version
+              = tablet_meta_.at(0)->tablet_meta_.report_status_.merge_snapshot_version_;
+          if ((minor_snapshot_version <= 0
+               && tablet_meta_.at(0)->tablet_meta_.table_store_flag_.with_major_sstable())
+              || (major_snapshot_version <= 0
+                  && tablet_meta_.at(1)->tablet_meta_.table_store_flag_.with_major_sstable())) {
+            ret = OB_ERR_UNEXPECTED;
+            STORAGE_LOG(WARN, "snapshot version is invalid", K(ret), K(minor_snapshot_version),
+                        K(major_snapshot_version));
+          } else if (major_snapshot_version < minor_snapshot_version) {
+            ret = OB_BACKUP_MAJOR_NOT_COVER_MINOR;
+            STORAGE_LOG(WARN, "major snapshot version is smaller than minor snapshot version",
+                        K(ret), K(major_snapshot_version), K(minor_snapshot_version));
+          }
+        }
+      } else if (3 == current_group.count()) {
```

Comment:
why sys tablet is 3?

Reply:
one ls have 3 ls_inner_tablet, is schedule together.